charts/supabase/templates/analytics/deployment.yaml

```diff
@@ -0,0 +1,154 @@
+{{- if .Values.analytics.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "supabase.analytics.fullname" . }}
+  labels:
+    {{- include "supabase.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.analytics.autoscaling.enabled }}
+  replicas: {{ .Values.analytics.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "supabase.analytics.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.analytics.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "supabase.analytics.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.analytics.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "supabase.analytics.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.analytics.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: init-db
```

Comment:
This is repeated in quite a few places, and probably could be in a couple of other deployments that don't currently have an init container db check. Could this be extracted to the top `_helpers.tpl` and `include`d?

Reply:
You are right. However, there might be slight changes to analytics `init-db` since changing `POSTGRES_BACKEND_URL` after initial deployment may cause it to return 404. I'm still trying to figure out where does the initial value resides and maybe add an alteration script to the `init-db`.

Reply:
If you are interested, I can try and adapt the `bitnami` way of doing things to what we have here. Personally, I think it would be worth exploring using the `bitnami` `postgresql` and `minio` charts as sub-dependencies. While they can be horribly slow updating some stuff, they are pretty good on the highly used charts, like postgres % co. What do you think?